sycl-fusion/jit-compiler/lib/KernelFusion.cpp

```diff
@@ -70,6 +70,50 @@ static bool isTargetFormatSupported(BinaryFormat TargetFormat) {
   }
 }
 
+extern "C" FusionResult materializeSpecConstants(
+    const char *KernelName, jit_compiler::SYCLKernelBinaryInfo &BinInfo,
+    std::vector<unsigned char> &SpecConstBlob, const std::string &TargetCPU,
+    const std::string &TargetFeatures) {
+  auto &JITCtx = JITContext::getInstance();
+
```

Comment:
I like that trick, and I did not know about `llvm::setCurrentDebugType` before, so thank you for that. My worry is though that this is too broad of an approach; if we ever decide to write more passes (which seem very likely) we will have no way of setting per-pass debugging output. I think I'd prefer to stick with `MATERIALIZER_DEBUG` macro.

Reply:
I had to tweak it a little bit, because `LLVM_DEBUG` [doesn't evaluate anything in our case](https://github.com/jchlanda/llvm/blob/jakub/jit_spec_const/llvm/include/llvm/Support/Debug.h#L64); so it needs to be first initialized. Let me know how you feel about it: https://github.com/jchlanda/llvm/commit/df9133fddf5a70354b6d4d84a271c7df5fc4627d#diff-d502135ccb3a4330d9335161290d1a1aac57062e4ac3c0a3acecc78fc683e2ddR282

Reply:
I understand. Good. I'll leave a comment with suggested changes. Good one!